devicetypes/sensative/sensative-strips-guard-700.src/sensative-strips-guard-700.groovy

```diff
@@ -0,0 +1,482 @@
+	logTrace "${cmd}"
+	
+	sendEventIfNew("firmwareVersion", (cmd.applicationVersion + (cmd.applicationSubVersion / 100)))
+}
+
+
+void zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+	int val = (cmd.batteryLevel == 0xFF ? 1 : safeToInt(cmd.batteryLevel))
+	if (val > 100) val = 100
+	if (val < 1) val = 1
+
+	String desc = "${device.displayName}: battery is ${val}%"
+	logDebug(desc)
+
+	sendEvent(name: "battery", value: val, unit: "%", isStateChange: true, descriptionText: desc)
+}
+
+
+void zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd) {
+	logTrace "${cmd}"
+
+	runIn(4, refreshSyncStatus)
+
+	Map param = configParams.find { it.num == cmd.parameterNumber }
+	if (param) {
+		logDebug "${param.name}(#${param.num}) = ${cmd.scaledConfigurationValue}"
+		setParamStoredValue(param.num, cmd.scaledConfigurationValue)
+	}
+	else {
+		logDebug "Unknown Parameter #${cmd.parameterNumber} = ${cmd.scaledConfigurationValue}"
+	}
+}
+
+
+void zwaveEvent(physicalgraph.zwave.commands.sensorbinaryv1.SensorBinaryReport cmd) {
+	logTrace "${cmd}"
+	sendContactEvent(cmd.sensorValue)
+}
+
+void zwaveEvent(physicalgraph.zwave.commands.notificationv3.NotificationReport cmd) {
+	logTrace "${cmd}"
+
+	switch (cmd.notificationType) {
+		case 6:
+			if ((cmd.event == 22) || (cmd.event == 23)) {
+				sendContactEvent(cmd.event == 22)
+			}
+			break
+			
+		case 7:
+			if ((cmd.event == 11) || (cmd.eventParameter[0] == 11)) {
+				sendTamperEvent(cmd.event == 11)
+			}
+			else if ((cmd.event == 2) || (cmd.eventParameter[0] == 2)) {
+				sendContactEvent(cmd.event == 2)
+			}
+			break
+	}
+}
+
+void sendContactEvent(rawVal) {
+	sendEventIfNew("contact", (rawVal ? "open" : "closed"))
+}
+
+void sendTamperEvent(rawVal) {
+	sendEventIfNew("tamper", (rawVal ? "detected" : "clear"))
+}
+
+
+void zwaveEvent(physicalgraph.zwave.Command cmd) {
+	logDebug "${cmd}"
+}
+
+
+void refreshSyncStatus() {
+	int changes = pendingChanges
+	sendEventIfNew("syncStatus", (changes ?  "${changes} Pending Changes" : "Synced"), false)
+}
+
+int getPendingChanges() {
+	return safeToInt(configParams.count { ((getSettingValue(it.num) != null) && (getSettingValue(it.num) != getParamStoredValue(it.num))) }) + ((state.wakeUpInterval != wakeUpIntervalSeconds) ? 1 : 0)
+}
+
+Integer getSettingValue(int paramNum) {
+	return safeToInt((settings ? settings["configParam${paramNum}"] : null), null)
+}
+
+Integer getParamStoredValue(int paramNum) {
+	return safeToInt(state["configVal${paramNum}"], null)
+}
+
+void setParamStoredValue(int paramNum, int value) {
+	state["configVal${paramNum}"] = value
+}
+
+
+void sendEventIfNew(String name, value, boolean displayed=true) {
+	String desc = "${device.displayName}: ${name} is ${value}"
+	if (device.currentValue(name) != value) {
+		if (name != "syncStatus") {
+			logDebug(desc)
+		}
+		sendEvent(name: name, value: value, descriptionText: desc, displayed: displayed)
+	}
+	else {
```

Comment:
Please remove this line break before the "else" statement.

Reply:
Removed